src/lib/crawler/crawler.ts

```diff
@@ -1,68 +1,222 @@
+  }
 
-    // loading fetched HTML into Cheerio object
+  // parses fetched HTML w/ Cheerio, makes array of MatchSummary objects with extracted info
+  extractMatchSummaries(html: string): MatchSummary[] {
     const $ = cheerio.load(html);
-    const matchIds: string[] = [];
+    const matchSummaries: MatchSummary[] = [];
 
     $("table#data-table-history tbody tr").each((_index, element) => {
-      const matchId = $(element).find("td:first-child").text().trim();
-      matchIds.push(matchId);
+      const matchId = $(element).find("td:nth-child(1)").text().trim();
+      const outcome = $(element).find("td:nth-child(3)").text().trim();
+      const points_change = $(element).find("td:nth-child(4)").text().trim(); //? todo?
+      const date = $(element).find("td:nth-child(5)").text().trim();
+      const duration = $(element).find("td:nth-child(6)").text().trim();
+      const arena = $(element).find("td:nth-child(7)").text().trim();
+      const teamBracketText = $(element)
+        .find("td:nth-child(2) a")
+        .text()
+        .trim();
+      const teamBracketRegex = /(.*?)\s*\((\d+v\d+)\)/;
+      let team_name = "";
+      let bracket = "";
+
+      if (teamBracketText) {
+        const teamBracketMatch = teamBracketText.match(teamBracketRegex);
+
+        if (teamBracketMatch) {
+          team_name = teamBracketMatch[1];
+          bracket = teamBracketMatch[2];
+        }
+      }
+      // creates object with extracted match data
+      const matchSummary: MatchSummary = {
+        matchId,
+        team_name,
+        date,
+        bracket,
+        arena,
+        points_change,
+        outcome,
+        duration,
+      };
+      // pushes each match's object to array
+      matchSummaries.push(matchSummary);
     });
 
-    return matchIds;
+    return matchSummaries;
   }
 
+  // calls fetchMatchHistoryHTML() & extractMatchSummaries()
+  async getMatchSummaries(params: {
+    character: string;
+    realm: string;
+  }): Promise<MatchSummary[]> {
+    const html = await this.fetchMatchHistoryHTML(params);
+    const matchSummaries = this.extractMatchSummaries(html);
+    return matchSummaries;
+  }
+
+  // fetches match data for a given match ID, returns data as object
+  async fetchMatchData(matchId: string): Promise<CharacterDetail[]> {
+    const response = await axios.post(
+      "https://armory.warmane.com/character/Dumpster/Blackrock/match-history",
+      `matchinfo=${matchId}`,
+      {
+        headers: {
+          "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8",
+          "User-Agent":
+            "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.5060.114 Safari/537.36",
+        },
+      }
+    );
+
+    return response.data;
+  }
+
+  // responsible for fetching match details using the match IDs of each given matchSummaries array, combining the details to create a single object, and restricting concurrent operations to 32
```

Comment:
This comment is actually more useful. I just don't like I have to scroll so far to read the whole thing.